src/AppInstallerRepositoryCore/Microsoft/Schema/1_4/DependenciesTable.cpp

```diff
@@ -0,0 +1,425 @@
+                {
+                    missingPackageNodes.emplace_back(dep);
+                    continue;
+                }
+
+                std::optional<SQLite::rowid_t> versionRowId;
+                if (depMinVersion.has_value())
+                {
+                    versionRowId = VersionTable::EnsureExists(connection, dep.MinVersion.value().ToString());
+                }
+
+                dependenciesTableRows.emplace_back(DependencyTableRow{ packageRowId.value(), manifestRowId, versionRowId });
+            }
+
+            ThrowOnMissingPackageNodes(missingPackageNodes);
+
+            StatementBuilder insertBuilder;
+            insertBuilder.InsertInto(s_DependenciesTable_Table_Name)
+                .Columns({ s_DependenciesTable_Manifest_Column_Name, s_DependenciesTable_MinVersion_Column_Name, s_DependenciesTable_PackageId_Column_Name })
+                .Values(Unbound, Unbound, Unbound);
+            SQLite::Statement insert = insertBuilder.Prepare(connection);
+
+            for (const auto& dep : dependenciesTableRows)
+            {
+                insert.Reset();
+                insert.Bind(1, dep.m_manifestRowId);
+                insert.Bind(2, dep.m_versionRowId.has_value() ? std::to_string(dep.m_versionRowId.value()) : "");
+                insert.Bind(3, dep.m_packageRowId);
+
+                insert.Execute(true);
+            }
+        }
+
+        std::optional<std::pair<SQLite::rowid_t, Utility::NormalizedString>> GetPackageRowIdAndVersion(SQLite::Connection& connection, const Manifest::Dependency& dependency)
+        {
+            auto packageId = IdTable::SelectIdByValue(connection, dependency.Id);
+
+            if (packageId.has_value())
+            {
+                return std::make_pair(packageId.value(), dependency.MinVersion.value().ToString());
+            }
+
+            return {};
+        }
+    }
+
+    std::string_view DependenciesTable::TableName()
+    {
+        return s_DependenciesTable_Table_Name;
+    }
+
+    void DependenciesTable::Create(SQLite::Connection& connection)
+    {
+        using namespace SQLite::Builder;
+        SQLite::Savepoint savepoint = SQLite::Savepoint::Create(connection, "createDependencyTable_v1_4");
+        constexpr std::string_view dependencyIndexByManifestId = "dependencies_manifest_id_index";
+        constexpr std::string_view dependencyIndexByPackageId = "dependencies_package_id_index";
+
+        StatementBuilder createTableBuilder;
+        createTableBuilder.CreateTable(TableName()).BeginColumns();
+        createTableBuilder.Column(IntegerPrimaryKey());
+
+        std::vector<DependenciesTableColumnInfo> notNullableDependenciesColumns =
+        {
+            { s_DependenciesTable_Manifest_Column_Name },
+            { s_DependenciesTable_MinVersion_Column_Name },
+        };
+
+        std::vector<DependenciesTableColumnInfo> nullableDependenciesColumns =
+        {
+            { s_DependenciesTable_PackageId_Column_Name }
```

Comment:
Looks like you swapped package id (which cannot be null) and min version (which can) in these.

General note on `vector`: since it requires allocations it isn't great for a fixed set of known values contained within a specific scope.  Either a `std::array` or implicit `initializer_list` in range-for would be more efficient.

```
for (auto& thing : { a, b, c })
    ...
```